python/master.py

```diff
@@ -0,0 +1,321 @@
+        tissue = " ".join(tissue)
+        tissue_list.append(tissue)
+
+    d = {'Celllines': cellline_list, 'Tissue' : tissue_list}
+    cell_df = pd.DataFrame(d)
+    cellline_tissue = cell_df.values
+    tissue_dict ={}
+    for data in cellline_tissue:
+       tissue_dict[data[0]] = data[1]
+    return tissue_dict, cellline_list
+
+def normalize(df, delete_column, cell_list):
+    """Takes in a dataframe with columns being celllines (with the exception of a column being genes or gene_ids) and normalizes the expression values.
+    
+    This will delete the desired column and then normalize the values and fill in the NaN values. It will then format the dataframe to back to how it originally looked. 
+    """
+
+    temp_df = df.copy()
+    
+    delete = temp_df[delete_column]
+    del temp_df[delete_column]
+    
+    x = temp_df.values.astype(float)
+    min_max_scaler = preprocessing.MinMaxScaler()
+    x_scaled = min_max_scaler.fit_transform(x)
+    df_normalized = pd.DataFrame(x_scaled)
+
+    df_normalized.insert(0, 'Gene', delete , True)
+    df_normalized.columns = ['Gene'] + cell_list
+    
+
+    #fill in NaN values
+    df_normalized = df_normalized.fillna(method='ffill')
+    return df_normalized
+
+def common(a,b):
+    """ This functions purpose is to find common elements of lists by converting each list to a set.
+    
+    The inputs of this function should both be lists. 
+    """
+    a_set = set(a)
+    b_set = set(b)
+    common = a_set.intersection(b_set) 
+    common = list(common)
+    return common 
+
+#Used to create correlation matrix
+def matrix(gene_list, histone_list, data1, data2):
+    """This functions purpose is to create two numpy matrices. The first value is a matrix of the R-values between the two data sets, and the second matrix is the corresponding p-values. 
+    
+    The gene-list and histone-list allow you to input big expression value data sets and then specific what genes and histones you want to use. 
+
+    This function is very specific. The dataframe created will contain histone markers as the columns and gene names as the index. 
+    
+    This function assumes you already took the intersection between the two arrays. If you do not do that, then the matrix dimensions will not agree, resulting in an error.
+    
+    The format for inputting matrix should have columns correspond to the celllines.
+    
+    """
+    Rmat = (len(gene_list), len(histone_list))
+    Rmat = np.zeros(Rmat)
+    Pmat = Rmat
+    i = 0 
+    for gene in gene_list:
+        j = 0
+        for histone in histone_list:
+            data1.loc[gene] = data1.loc[gene].fillna(method='ffill')
+            data2.loc[histone] = data2.loc[histone].fillna(method='ffill')
+            correlation = pearsonr(data1.loc[gene], data2.loc[histone])
+            Rmat[i][j] = correlation[0]
+            Pmat[i][j]
+            j = j+1
+        i = i+1
+    return Rmat, Pmat
+    
+def pearson_dfs(gene_list, histone_list, data1, data2):
```

Comment:
This code seems redundant with the `matrix` function, which is also not a great way to name the function, since matrix doesn't tell me anything about the functionality. You may want to merge the two functions together. 

If you really need to keep the numpy array, make another argument called dtype, and make it a conditional, such that if dtype='numpy', return numpy array, else return pandas dataframe.